### PR TITLE
feat: new modules, fixes, and error handling

### PR DESCRIPTION
**Pull request by agustinustheo**
### JIRA LINK
- https://blocksphere2020.atlassian.net/browse/DBIO-2070
- https://blocksphere2020.atlassian.net/browse/DBIO-2071
- https://blocksphere2020.atlassian.net/browse/DBIO-2072
- https://blocksphere2020.atlassian.net/browse/DBIO-2073
- https://blocksphere2020.atlassian.net/browse/DBIO-2112
- https://blocksphere2020.atlassian.net/browse/DBIO-2114
- https://blocksphere2020.atlassian.net/browse/DBIO-2115
- https://blocksphere2020.atlassian.net/browse/DBIO-2118
- https://blocksphere2020.atlassian.net/browse/DBIO-2556
- https://blocksphere2020.atlassian.net/browse/DBIO-2575
- https://blocksphere2020.atlassian.net/browse/DBIO-2594

### Changelog
- Fix typo.
- Fix type.
- Error handling using promises.
- Doctor certifications pallet and integration tests.
- Hospital certifications pallet and integration tests.

### CODE REVIEW

**Comment by agustinustheo** [comment]
Waiting for updates from @azmimuhammad

**Comment by agustinustheo** [comment]
image
Tests successful

**Comment by agustinustheo** [comment]
image

All successful. No errors for now

**Review by JackyRahman — APPROVED** [review]

File: src/command/labs/index.ts [hunk 1]
```diff
@@ -42,74 +61,183 @@ export async function updateLabVerificationStatus(
...
+        events,
+        status,
+        callback,
+        resolve,
+        reject,
+        unsub,
+      } as ExtrinsicCallbackParameters);
+    });
   });
 }
```
Comment (ciniiia) [inline src/command/labs/index.ts:132]: please add functions for stakeLabFee and unstakeLabFee

File: src/models/genetic-analysts/genetic-analysis-orders/index.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export class GeneticAnalysisOrder {
     this.customerBoxPublicKey = anyJson.customerBoxPublicKey;
     this.sellerId = anyJson.sellerId;
     this.geneticDataId = anyJson.geneticDataId;
-    this.geneticAnalysisdTrackingId = anyJson.geneticAnalysisTrackingId;
+    this.geneticAnalysisIdTrackingId = anyJson.geneticAnalysisTrackingId;
```
Comment (ciniiia) [inline src/models/genetic-analysts/genetic-analysis-orders/index.ts]: typo ? `geneticAnalysisTrackingId`

File: src/models/genetic-analysts/genetic-analysis-orders/index.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ export class GeneticAnalysisOrder {
   customerBoxPublicKey: string;
   sellerId: string;
   geneticDataId: string;
-  geneticAnalysisdTrackingId: string;
+  geneticAnalysisIdTrackingId: string;
```
Comment (ciniiia) [inline src/models/genetic-analysts/genetic-analysis-orders/index.ts]: typo ? `geneticAnalysisTrackingId`

File: test/e2e/genetic-analysts/genetic-analysis-order.spec.ts [hunk 4]
```diff
@@ -144,15 +144,15 @@ describe('Genetic Analysis Order Pallet Integration Tests', () => {
     await submitGeneticAnalysis(
       api, 
       pair, 
-      geneticAnalysisOrder.geneticAnalysisdTrackingId, 
+      geneticAnalysisOrder.geneticAnalysisIdTrackingId,
```
Comment (ciniiia) [inline test/e2e/genetic-analysts/genetic-analysis-order.spec.ts]: `geneticAnalysisTrackingId`

File: test/e2e/genetic-analysts/genetic-analysis-order.spec.ts [hunk 5]
```diff
@@ -144,15 +144,15 @@ describe('Genetic Analysis Order Pallet Integration Tests', () => {
...
+      geneticAnalysisOrder.geneticAnalysisIdTrackingId, 
       geneticDataMock.reportLink,
       "string"
     );
 
     await processGeneticAnalysis(
       api,
       pair,
-      geneticAnalysisOrder.geneticAnalysisdTrackingId, 
+      geneticAnalysisOrder.geneticAnalysisIdTrackingId,
```
Comment (ciniiia) [inline test/e2e/genetic-analysts/genetic-analysis-order.spec.ts]: `geneticAnalysisTrackingId`

File: test/e2e/genetic-analysts/genetic-analysis-order.spec.ts [hunk 6]
```diff
@@ -186,15 +186,15 @@ describe('Genetic Analysis Order Pallet Integration Tests', () => {
     await submitGeneticAnalysis(
       api, 
       pair, 
-      _geneticAnalysisOrder.geneticAnalysisdTrackingId, 
+      _geneticAnalysisOrder.geneticAnalysisIdTrackingId,
```
Comment (ciniiia) [inline test/e2e/genetic-analysts/genetic-analysis-order.spec.ts]: `geneticAnalysisTrackingId`

File: test/e2e/genetic-analysts/genetic-analysis-order.spec.ts [hunk 7]
```diff
@@ -186,15 +186,15 @@ describe('Genetic Analysis Order Pallet Integration Tests', () => {
...
+      _geneticAnalysisOrder.geneticAnalysisIdTrackingId, 
       geneticDataMock.reportLink,
       "comment"
     );
 
     await processGeneticAnalysis(
       api,
       pair,
-      _geneticAnalysisOrder.geneticAnalysisdTrackingId, 
+      _geneticAnalysisOrder.geneticAnalysisIdTrackingId,
```
Comment (ciniiia) [inline test/e2e/genetic-analysts/genetic-analysis-order.spec.ts]: `geneticAnalysisTrackingId`

File: test/e2e/genetic-analysts/genetic-analysis.spec.ts [hunk 8]
```diff
@@ -101,8 +101,8 @@ describe('Genetic Analysis Pallet Integration Tests', () => {
     expect(geneticAnalysisOrder.customerBoxPublicKey).toEqual(geneticAnalysisOrdersDataMock.customerBoxPublicKey);
 
     const submitGeneticAnalysisPromise: Promise<GeneticAnalysis> = new Promise((resolve, reject) => { // eslint-disable-line
-      submitGeneticAnalysis(api, pair, geneticAnalysisOrder.geneticAnalysisdTrackingId, geneticDataMock.reportLink, "string", () => {
-        queryGeneticAnalysisByGeneticAnalysisTrackingId(api, geneticAnalysisOrder.geneticAnalysisdTrackingId)
+      submitGeneticAnalysis(api, pair, geneticAnalysisOrder.geneticAnalysisIdTrackingId, geneticDataMock.reportLink, "string", () => {
```
Comment (ciniiia) [inline test/e2e/genetic-analysts/genetic-analysis.spec.ts]: `geneticAnalysisTrackingId`

File: test/e2e/genetic-analysts/genetic-analysis.spec.ts [hunk 9]
```diff
@@ -101,8 +101,8 @@ describe('Genetic Analysis Pallet Integration Tests', () => {
...
+        queryGeneticAnalysisByGeneticAnalysisTrackingId(api, geneticAnalysisOrder.geneticAnalysisIdTrackingId)
```
Comment (ciniiia) [inline test/e2e/genetic-analysts/genetic-analysis.spec.ts]: `geneticAnalysisTrackingId`

**Review by ciniiia — CHANGES_REQUESTED** [review]
Please add functions for `stakeLabFee` and `unstakeLabFee` and change `geneticAnalysisIdTrackingId` to `geneticAnalysisTrackingId`

**Review by ciniiia — APPROVED** [review]
LGTM!

**Comment by agustinustheo** [comment]
image

All tests are passing again.

**Pull request merged**